java/test/jmri/jmrit/display/layoutEditor/LayoutEditorChecksTest.java

```diff
@@ -1,40 +1,349 @@
+        Assert.assertEquals("resultsMenuItem0.getText(): ", myBlockName, resultsMenuItem.getText());
+
+        JMenuOperator resultsJMO = new JMenuOperator((JMenu) resultsMenuItem);
+        JPopupMenu resultsPopupMenu = resultsJMO.getPopupMenu();
+        Assert.assertEquals("Correct number of check noncontiguous blocks results submenu items",
+                2, resultsPopupMenu.getSubElements().length);
+        JMenuItem resultsSubMenuItem = (JMenuItem) resultsPopupMenu.getComponent(0);
+        Assert.assertEquals("resultsSubMenuItem0.getText(): ", myBlockName + ":#1", resultsSubMenuItem.getText());
+        resultsSubMenuItem = (JMenuItem) resultsPopupMenu.getComponent(1);
+        Assert.assertEquals("resultsSubMenuItem1.getText(): ", myBlockName + ":#2", resultsSubMenuItem.getText());
+    }   //testCheckNonContiguousBlocks
+
+    @Test
+    public void testCheckUnnecessaryAnchors() {
+        Assume.assumeFalse(GraphicsEnvironment.isHeadless());
+        Assert.assertNotNull("toolsMenuTitle not null", toolsMenuTitle);
+        JMenuOperator toolsJMO = new JMenuOperator(layoutEditorEFO, toolsMenuTitle);
+        Assert.assertNotNull("CheckMenuTitle not null", checkMenuTitle);
+        Assert.assertNotNull("checkUnnecessaryAnchorsMenuTitle not null", checkUnnecessaryAnchorsMenuTitle);
+
+        JPopupMenu toolsPopupMenu = toolsJMO.getPopupMenu();
+        JMenuItem checkMenuItem = (JMenuItem) toolsPopupMenu.getComponent(0);
+        Assert.assertEquals("checkMenuItem.getText(): ",
+                checkMenuTitle, checkMenuItem.getText());
+
+        JMenuOperator checkJMO = new JMenuOperator((JMenu) checkMenuItem);
+        JPopupMenu checksPopupMenu = checkJMO.getPopupMenu();
+        JMenuItem checkUnnecessaryAnchorsMenuItem = (JMenuItem) checksPopupMenu.getComponent(3);
+        Assert.assertEquals("subMenuItem.getText(): ",
+                checkUnnecessaryAnchorsMenuTitle, checkUnnecessaryAnchorsMenuItem.getText());
+
+        JMenuOperator checkUnnecessaryAnchorsJMO = new JMenuOperator((JMenu) checkUnnecessaryAnchorsMenuItem);
+        checkUnnecessaryAnchorsJMO.doClick();
+
+        JPopupMenu checkUnnecessaryAnchorsPopupMenu = checkUnnecessaryAnchorsJMO.getPopupMenu();
+
+        //verify results
+        Assert.assertEquals("Correct number of check unnecessary anchors results menu items",
+                2, checkUnnecessaryAnchorsPopupMenu.getSubElements().length);
+        JMenuItem resultsMenuItem = (JMenuItem) checkUnnecessaryAnchorsPopupMenu.getComponent(0);
+        Assert.assertEquals("resultsMenuItem0.getText(): ", a1.getName(), resultsMenuItem.getText());
+        resultsMenuItem = (JMenuItem) checkUnnecessaryAnchorsPopupMenu.getComponent(1);
+        Assert.assertEquals("resultsMenuItem1.getText(): ", a2.getName(), resultsMenuItem.getText());
+    }   //testCheckUnnecessaryAnchors
+
+    @BeforeClass
+    public static void setUpClass() {
```

Comment:
I strongly feel that this setup should be done in a per-test setup method rather than in a static method.

It is too easy to have one test contaminate the environment of a following test if they are all using the same configuration.

Reply:
The items setup here should not be effected by any tests… if they are then that's a failure and I want to catch it. Doing this before every test would hide that error and also just waste time. (The LayoutEditorChecks shouldn't effect anything: they only report possible errors.)